pkg/registry/core/shoot/storage/binding.go

```diff
@@ -116,6 +116,7 @@ func (r *BindingREST) assignShoot(ctx context.Context, shootUID types.UID, shoot
 			return nil, fmt.Errorf("shoot %s is being deleted, cannot be assigned to a seed", shoot.Name)
 		}
 
+		shoot.Generation++
```

Comment:
A unit test for this would be great, this seems to be crucial and easy to miss :)